pageserver/src/repository.rs

```diff
@@ -96,24 +107,26 @@ pub trait Timeline: Send + Sync {
     fn history<'a>(&'a self) -> Result<Box<dyn History + 'a>>;
 }
 
-pub trait History: Iterator<Item = Result<RelationUpdate>> {
+pub trait History: Iterator<Item = Result<Modification>> {
     /// The last_valid_lsn at the time of history() call.
     fn lsn(&self) -> Lsn;
 }
 
 #[derive(Debug, PartialEq, Eq, Serialize, Deserialize)]
-pub struct RelationUpdate {
-    pub rel: RelTag,
+pub struct Modification {
+    pub tag: ObjectTag,
     pub lsn: Lsn,
-    pub update: Update,
+    pub data: Vec<u8>,
 }
 
-#[derive(Debug, PartialEq, Eq, Serialize, Deserialize)]
-pub enum Update {
-    Page { blknum: u32, img: Bytes },
-    WALRecord { blknum: u32, rec: WALRecord },
-    Truncate { n_blocks: u32 },
-    Unlink,
+impl Modification {
+    pub fn new(entry: (ObjectTag, Lsn, Vec<u8>)) -> Modification {
+        Modification {
+            tag: entry.0,
+            lsn: entry.1,
+            data: entry.2,
+        }
+    }
```

Comment:
Why is the input a tuple? Why not just take regular fn parameters?

Reply:
Sorry, it is my porting of Patric's code (implementation of branch push). I just need to switch to new object tag model and make copying of data at raw level (just copy object storage records rather then replaying high level operations). I tried to minimize number of changes in the code and did only which are needed. I didn't perform any refactoring.

I think that tuple is passed to new method just because it is returned by object iterator.